crates/runtime/src/datafusion.rs

```diff
@@ -225,13 +225,33 @@ impl DataFusion {
                 }
             }
 
+            let mut attempts = 1;
+            loop {
+                if ctx
+                    .state()
+                    .statement_to_plan(statements[0].clone())
+                    .await
+                    .is_ok()
+                {
+                    break;
+                }
+
+                if attempts == 10 {
+                    tracing::error!("Unable to generate plan for view, retrying...");
+                    return;
+                }
+                attempts += 1;
+                sleep(Duration::from_secs(1)).await;
+            }
```

Comment:
This is the wrong fix - we need to understand why the planning errors happen and make the code above this detect that condition explicitly.

Reply:
All of the planning errors we saw before were because the dependent tables didn't exist, but we're checking for that explicitly. What is the new error?

Reply:
For context, this was the spicepod and error
```yaml
kind: Spicepod
name: demo
version: v1beta1
 
datasets:
- from: spice.ai/btc.blocks
  name: btc_blocks
  acceleration:
    refresh_mode: full
- name: test_view
  sql: |
    SELECT ts, CAST(y as DOUBLE) AS y, CAST(y2 as DOUBLE) as y2
    FROM (
          SELECT
              ROW_NUMBER() OVER (ORDER BY a.number) AS ts,
              2 * ROW_NUMBER() OVER (ORDER BY a.number) + 50 AS y,
              -3 * ROW_NUMBER() OVER (ORDER BY a.number) + 1 AS y2
          FROM
              btc_blocks a
          ORDER BY ts DESC
          LIMIT 50
    ) nested_0;
 
models:
  - name: demo
    from: spice.ai/Jeadie/my-first-app/darts@latest
    model_type: onnx
    datasets:
      - test_view
```